backend/models/group.js

```diff
@@ -132,19 +141,86 @@ groupSchema.statics.findIfcGroupByUID = function(dbCol, uid){
 				if (groupObjectsMap && groupObjectsMap.length > 0) {
 					group.objects = [];
 					for (let id in groupObjectsMap) {
-						group.objects.push(groupObjectsMap[id]);
+						if (groupObjectsMap.hasOwnProperty(id)) {
```

Comment:
Shouldn't it be `if (id)`?

Reply:
I think it's this way because of the linter James is using that you have to explicitly check the groupObjectsMap has the field.

I suggest if statements don't need to be there and we do not impose this linting rule. Makes no sense to me

Reply:
Why is TSLint enabled when it's a js script?

Reply:
I think linting should stick to tidying code instead of depicting what bugs to catch though. We are adding an extra branch prediction here (and in other places) for not a lot of reason.

Reply:
I vote for turning it off.  Charence as well